ksa_compliance/standard_doctypes/sales_invoice_item.py

```diff
@@ -1,5 +1,11 @@
 import frappe
+import json
 
 
 def calculate_tax_amount(self, method):
-    print(self.as_dict())
+    if self.items:
+        for item in self.items:
+            item_tax_rate = json.loads(item.item_tax_rate)["Expenses Included In Valuation - KD"]
```

Comment:
There are a couple of problems with this:
* We're running this hook on every `Sales Invoice` save, and parsing JSON for every item in the invoice. This might slow down invoice saving significantly (we'd have to measure), and it's already slow to begin with. Invoice saving is often a bottleneck for our clients, so we like to keep it as fast as possible
* You're hard-coding a reference to the tax template name. This won't work in the general case, and it doesn't fail gracefully so it'll halt invoice saving if the tax template is named differently